demo/src/main/java/com/project/demo/rest/auth/AuthRestController.java

```diff
@@ -62,14 +66,33 @@ public ResponseEntity<LoginResponse> authenticate(@RequestBody User user) {
     @PostMapping("/signup")
     public ResponseEntity<?> registerUser(@RequestBody User user) {
         user.setPassword(passwordEncoder.encode(user.getPassword()));
-        Optional<Role> optionalRole = roleRepository.findByName(RoleEnum.USER);
 
+        Optional<Role> optionalRole = roleRepository.findByName(RoleEnum.USER);
         if (optionalRole.isEmpty()) {
-            return null;
+            return ResponseEntity.badRequest().body("Role not found");
         }
         user.setRole(optionalRole.get());
+
+        if (user.getCountry() != null && user.getCountry().getCountryId() != null) {
+            Optional<Country> optionalCountry = countryRepository.findById(user.getCountry().getCountryId());
+            if (optionalCountry.isPresent()) {
+                user.setCountry(optionalCountry.get());
+            } else {
+                // Crear y guardar una nueva entidad Country si no existe
+                Country country = new Country();
+                country.setCountryId(user.getCountry().getCountryId());
+                country.setCountryName("Default Country Name");
+                country.setCountryCode("Default Country Code");
+                country.setCurrency(new Currency());
+                country.setOperational(true);
+                country = countryRepository.save(country);
```

Comment:
Is it safe to have the repository call here since Juan changed the architecture?

Reply:
could be deleted